File: bolt-servlet/src/test/java/util/ResourceLoader.java
package util;

import com.google.gson.Gson;
import com.google.gson.JsonElement;
import com.google.gson.JsonObject;
import com.slack.api.bolt.AppConfig;
import lombok.extern.slf4j.Slf4j;
import samples.DialogSample;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.util.HashMap;
import java.util.Map;

import static java.util.stream.Collectors.joining;

@Slf4j
public class ResourceLoader {

    private ResourceLoader() {
    }

    public static AppConfig loadAppConfig() {
        AppConfig config = new AppConfig();
        ClassLoader classLoader = DialogSample.class.getClassLoader();
        // https://github.com/slackapi/java-slack-sdk/blob/master/bolt/src/test/resources
        try (InputStream is = classLoader.getResourceAsStream("appConfig.json");
             InputStreamReader isr = new InputStreamReader(is)) {
            String json = new BufferedReader(isr).lines().collect(joining());
            JsonObject j = new Gson().fromJson(json, JsonElement.class).getAsJsonObject();
            config.setSigningSecret(j.get("signingSecret").getAsString());
            config.setSingleTeamBotToken(j.get("singleTeamBotToken").getAsString());
        } catch (IOException e) {
            log.error(e.getMessage(), e);
        }
        return config;
    }

    public static Map<String, String> loadValues() {
        ClassLoader classLoader = DialogSample.class.getClassLoader();
        // src/test/resources
        try (InputStream is = classLoader.getResourceAsStream("appConfig.json");
             InputStreamReader isr = new InputStreamReader(is)) {
            String json = new BufferedReader(isr).lines().collect(joining());
            return new Gson().fromJson(json, HashMap.class);
        } catch (IOException e) {
            log.error(e.getMessage(), e);
            throw new RuntimeException(e);
        }
    }

    public static String load(String filepath) {
        ClassLoader classLoader = DialogSample.class.getClassLoader();
        try (InputStream is = classLoader.getResourceAsStream(filepath);
             InputStreamReader isr = new InputStreamReader(is)) {
            return new BufferedReader(isr).lines().collect(joining());
        } catch (IOException e) {
            log.error(e.getMessage(), e);
        }
        return null;
    }

}

File: bolt-kotlin-examples/src/main/kotlin/util/ResourceLoader.kt
package util

import com.google.gson.Gson
import com.google.gson.JsonElement
import com.slack.api.bolt.AppConfig
import org.slf4j.LoggerFactory
import java.io.BufferedReader
import java.io.IOException
import java.io.InputStreamReader
import java.util.stream.Collectors.joining

class ResourceLoader {

    companion object {
        private val logger = LoggerFactory.getLogger("resource-loader")

        @Suppress("DEPRECATION")
        fun loadAppConfig(name: String = "appConfig.json"): AppConfig {
            val config = AppConfig()
            val classLoader = ResourceLoader::class.java.classLoader
            try {
                classLoader.getResourceAsStream(name).use { resource ->
                    InputStreamReader(resource).use { isr ->
                        val json = BufferedReader(isr).lines().collect(joining())
                        val j = Gson().fromJson(json, JsonElement::class.java).asJsonObject
                        config.signingSecret = j.get("signingSecret").asString
                        config.verificationToken = j.get("verificationToken")?.asString
                        config.singleTeamBotToken = j.get("singleTeamBotToken")?.asString
                        config.clientId = j.get("clientId")?.asString
                        config.clientSecret = j.get("clientSecret")?.asString
                        config.redirectUri = j.get("redirectUri")?.asString
                        config.scope = j.get("scope")?.asString
                        config.userScope = j.get("userScope")?.asString
                        config.oauthStartPath = j.get("oauthStartPath")?.asString
                        config.oauthCallbackPath = j.get("oauthCallbackPath")?.asString
                        config.oauthCompletionUrl = j.get("oauthCompletionUrl")?.asString
                        config.oauthCancellationUrl = j.get("oauthCancellationUrl")?.asString
                        config.isOAuthStartEnabled = config.oauthStartPath != null
                        config.isOAuthCallbackEnabled = config.oauthCallbackPath != null
                    }
                }
            } catch (e: Throwable) {
                logger.error(e.message, e)
                throw e
            }
            return config
        }

        fun load(filepath: String): String? {
            return try {
                val classLoader = ResourceLoader::class.java.classLoader
                classLoader.getResourceAsStream(filepath).use { resource ->
                    InputStreamReader(resource).use { isr ->
                        BufferedReader(isr).lines().collect(joining())
                    }
                }
            } catch (e: IOException) {
                logger.error(e.message, e)
                null
            }
        }

    }

}